src/app/shared/components/addregressionregion/addregressionregion.component.ts

```diff
@@ -0,0 +1,422 @@
+    this.modalRef = this._modalService.open(this.modalElement, { backdrop: 'static', keyboard: false, size: 'lg' });
+    this.modalRef.result.then(
+      result => {
+        // this is the solution for the first modal losing scrollability
+        if (document.querySelector('body > .modal')) {
+          document.body.classList.add('modal-open');
+        }
+        this.CloseResult = `Closed with: ${result}`;
+      },
+      reason => {
+        this.CloseResult = `Dismissed ${this.getDismissReason(reason)}`;
+        this.cancelCreateRegression();
+      }
+    );
+  }
+
+  outputWimMessages(msg) {
+    // output messages from http request to toast
+    for (const key of Object.keys(msg)) {
+      for (const item of msg[key]) {
+        this._toasterService.pop(key, key.charAt(0).toUpperCase() + key.slice(1), item);
+      }
+    }
+  }
+
+  public showAddRegRegion() {
+    const addRegRegForm: AddRegressionRegion = {
+      show: true,
+      regRegionID: null
+  }
+    this._nssService.setAddRegressionRegionModal(addRegRegForm);
+  }
+
+  private getDismissReason(reason: any): string {
+    if (reason === ModalDismissReasons.ESC) return 'by pressing ESC';
+    else if (reason === ModalDismissReasons.BACKDROP_CLICK) return 'by clicking on a backdrop';
+    else return `with: ${reason}`;
+  }
+
+  ngOnDestroy() {
+    this.modalSubscript.unsubscribe();
+  }
+
+
+  // show add regression region modal
+  public showNewRegressionRegionForm(rr?) {
+    // shows form for creating new regression or editing regression 
+    if (rr) { // edit existing regression region
+    this._loaderService.showFullPageLoad();
+    this.addRegReg = false;
+    this._settingsService.getEntities(this.configSettings.regRegionURL + '/' + rr + '?getpolygon=true')
+      .subscribe((res) => {
+        this.selectedRegRegion = res;
+        this.newRegRegForm.controls['name'].setValue(this.selectedRegRegion.name);
+        this.newRegRegForm.controls['description'].setValue(this.selectedRegRegion.description);
+        this.newRegRegForm.controls['code'].setValue(this.selectedRegRegion.code);
+        this.newRegRegForm.controls['location'].setValue(this.selectedRegRegion.location);
+        if (this.selectedRegRegion.citationID) {
+          this.newRegRegForm.controls['citationID'].setValue(this.selectedRegRegion.citationID);
+          this.addCitation = true;
+          this._settingsService.getEntities(this.configSettings.citationURL + '/' + this.selectedRegRegion.citationID)
+          .subscribe(res => {
+              this.selectedCitation = res;
+              this.newCitForm.controls['title'].setValue(this.selectedCitation.title);
+              this.newCitForm.controls['author'].setValue(this.selectedCitation.author);
+              this.newCitForm.controls['citationURL'].setValue(this.selectedCitation.citationURL);
+          });
+        } else {
+          this.addCitation = false;
+        }
+        if (this.selectedRegRegion.location) {
```

Comment:
I think adding some comments or extra lines in this whole `showNewRegressionRegionForm` function would help improve readability, especially in places like this.